### ISSUE TITLE
Canceled purchases, doen't appears in inventory.GetAllPurchases() list

### ISSUE THREAD

**Issue by dremecker** [onepf/OpenIAB#461]
Hello,

I'm doing some test with google in app billing, and i've got an issue with canceled purchases. When i cancel a purchase, after a while, purchase doesn't appear in inventory.GetAllPurchases() list.

I'm trying to catch purchase state (0/1/2) to determined if a purchased product have been refunded or canceled. Am I doing wrong ?

For this test, I'm using google test account which can test purchase with my app SKU.

My debug code (inside query inventory succeeded event) :

			/* DEBUG START */

			List<Purchase> pList = inventory.GetAllPurchases();

			foreach (Purchase p in pList) {
				Debug.Log(p.Sku + " (" + p.OrderId + ") : " + p.PurchaseState);
			}

			/* DEBUG END */

Log entry when it works (so, only when purchased) : 
03-09 11:54:40.421: I/Unity(7564): sku_premium (12999763169054705758.1388484591254215) : 0

Regards,

**Comment by RomanZhilich** [comment]
Hello.
Looks like others have the same problem with Google Billing:
* https://stackoverflow.com/questions/19227676/android-in-app-billing-refund-cancel-takes-long
* https://stackoverflow.com/questions/14303850/android-in-app-billing-purchase-state-stays-purchased-after-order-cancelation

Also I'm pretty sure canceled purchases are not meant to stay in user inventory forever.
However it is, I don't think this issue is related to this library, since we only wrap Google Play response.

**Comment by dremecker** [comment]
I see those posts before and i'm aware of google caching methods. My issue is not exactly the same.
When i clear google play cache, i can retreive inventory. For exemple, if i purchase an item on an other device, i can get purchase in inventory.GetAllPurchases() list with purchaseState = 0. It's only work, for me, with purchased item not with canceled ones.

I guess the issue comes from google and his testing policies / methods that not allow us to cancel a purchase without erase it from purchase list, but i have no clues.

I'm trying to do some tests with real charged purchases. I'll come back to you with the results when it's done.

**Comment by dremecker** [comment]
Hello,

As promised I come back to you with more informations about my last tests. Even if you do a real purchase (with charge) the purchase state doesn't change when you refund it nor when you cancel it. 

Purchase state is alway 0 if you purchase an item but when you refund or cancel it, it erases purchase entry from purchase list like the pucharse action doesn't happen.

**Comment by RomanZhilich** [comment]
You suspect our library has something to do with this behavior?
Our implementation mostly reuses code from official Google sample.

**Comment by dremecker** [comment]
Yes I think but I'm not sure, as I don't have the expertise to check your library.

I did some Internet search and I found nothing about this kind of issue with google lib. That is why I think your library (or, at least, your unity implementation) could has something to do with this.

**Comment by RomanZhilich** [comment]
I suspect it might be related to ```isValidDataSignature() ``` as you mentioned in #464. Please try it again with ```VERIFY_SKIP``` flag.

**Comment by dremecker** [comment]
Hello,

It doen't change anything for me. After my last tests, I notice that if purchase is refunded, the purchase state is still 0 (like when purchase is puchased), if purchase canceled purchase list is always empty. 

From my point of view, it seems that purchase state (and list) match with owned sku list. Maybe it's just a coincidence.

**Comment by makrana** [comment]
Hello,

I am also stuck in same issue. For me even if subscription is cancelled from Google play, I get purchased state as 0. And if I re-subscribe before 1st subscription ends, still I get the data of cancelled subscription and not for the new subscription.
